lib/src/git_backend.rs

```diff
@@ -941,40 +983,60 @@ impl Backend for GitBackend {
                 parents.push(validate_git_object_id(parent_id)?);
             }
         }
+        let parents: SmallVec<[gix::ObjectId; 1]> = parents.into();
         let extras = serialize_extras(&contents);
+
         // If two writers write commits of the same id with different metadata, they
         // will both succeed and the metadata entries will be "merged" later. Since
         // metadata entry is keyed by the commit id, one of the entries would be lost.
         // To prevent such race condition locally, we extend the scope covered by the
         // table lock. This is still racy if multiple machines are involved and the
         // repository is rsync-ed.
         let (table, table_lock) = self.read_extra_metadata_table_locked()?;
-        let id = loop {
-            let git_id = locked_repo
-                .commit_as(
-                    committer,
-                    author,
-                    create_no_gc_ref(),
-                    message,
-                    git_tree_id,
-                    parents.iter().copied(),
-                )
-                .map_err(|err| BackendError::WriteObject {
-                    object_type: "commit",
-                    source: Box::new(err),
-                })?;
-            let id = CommitId::from_bytes(git_id.as_bytes());
-            match table.get_value(id.as_bytes()) {
+        let (git_id, commit) = loop {
+            let commit = gix::objs::Commit {
+                message: message.to_owned().into(),
```

Comment:
nit: (at your option) `message` could be borrowed as `contents.description.as_str()` to remove `.to_owned()`

Reply:
sure, thanks